lib/ai/prompt.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ai
+
+import "fmt"
+
+const promptSummarizeTitle = `You will be given a message. Create a short summary of that message.
+Respond only with summary, nothing else.`
+
+const initialAIResponse = `Hey, I'm Teleport - a powerful tool that can assist you in managing your Teleport cluster via ChatGPT.`
+
+const promptExtractInstruction = `If the input is a request to complete a task on a server, try to extract the following information:
+- A Linux shell command
+- One or more target servers
+- One or more target labels
+
+If there is a lack of details, provide most logical solution.
+Ensure the output is a valid shell command.
+There must be at least one target server or label, otherwise we do not have enough information to complete the task.
+Provide the output in the following format with no other text:
+
+{
+	"command": "<command to run>",
+	"nodes": ["<server1>", "<server2>"],
+	"labels": [
+		{
+			"key": "<label1>",
+			"value": "<value1>",
+		},
+		{
+			"key": "<label2>",
+			"value": "<value2>",
+		}
+	]
+}
+
+If the user is not asking to complete a task on a server - disgard this entire message and respond as part of a normal conversation instead.`
+
+func promptCharacter(username string) string {
+	return fmt.Sprintf(`You are Teleport, a tool that users can use to connect to Linux servers and run relevant commands, as well as have a conversation.
+A Teleport cluster is a connectivity layer that allows access to a set of servers. Servers may also be referred to as nodes.
+Nodes sometimes have labels such as "production" and "staging" assigned to them. Labels are used to group nodes together.
```

Comment:
Interesting way to describe labels. I know almost nothing about this stuff, but wouldn't it make more sense to describe labels as being part of the RBAC system and less about how to organize groups of nodes?

Reply:
Ideally yes. However GPT-4 in this PR knows very little about the inner workings of Teleport as it doesn't have a knowledge source from a vector db to refer to (coming soon). I attempted that explanation at first but it was too complicated for it to grasp without more data and yielded less reliable results.